flow360/component/simulation/framework/entity_registry.py

```diff
@@ -85,7 +85,9 @@ def get_bucket(self, by_type: EntityBase) -> EntityRegistryBucket:
             by_type.model_fields["private_attribute_registry_bucket_name"].default,
         )
 
-    def find_by_naming_pattern(self, pattern: str) -> list[EntityBase]:
+    def find_by_naming_pattern(
+        self, pattern: str, enforce_output_as_list: bool = True, error_when_no_match: bool = False
+    ) -> list[EntityBase]:
         """
         Finds all registered entities whose names match a given pattern.
```

Comment:
what "pattern" formats do we support? Is it glob or regex? Would be good to include some examples

Reply:
glob or regex. We can explain better later.